### PR TITLE
[MacOS]Working input number rendered

### PR DESCRIPTION
**Pull request by mukuagar**
## Related Issue
Please use one of the well-known [github fixes keywords](https://help.github.com/en/articles/closing-issues-using-keywords) to reference
the issues fixed with this PR (eg Fixes #<github issue number>). If an issue doesn't yet exist please create one if reasonable to aid 
in issue tracking.

**NOTE**: For multiple issues resolved by this PR use the corresponding keywords **every time** in a comma-delimited list per the reference
page above.

## Description
Create a [descriptive title and provide all of the relevant details in the description](https://chris.beams.io/posts/git-commit/). 
This information lives with the code and will help future readers (***including yourself***) and will serve as documentation.

At the very least please describe the issue you are addressing, and what your fix entails. 

This PR contains:
* UI changes for InputNumberField
* Functional changes to allow users to input "."(dot) and submit decimal numbers

## Sample Card
If appropriate, please include a link to a card in one of the samples directories that can be used to validate this change. This can be an existing card or a card added with this PR.

Light theme 
image

Dark theme
image

Floating point Numbers:
image
image
image

## How Verified
How you verified the fix, including one or all of the following: 
1. New unit tests that were added if any. If none were added please add a quick line explaining why not.
2. Existing relevant unit/regression tests that you ran
3. Manual scenario verification if any; ***Do include .gif's or screenshots of the testing you performed here if you think that it 
will aid in code reviews or corresponding fixes on other platforms for eg.***

New tests added

### CODE REVIEW

File: source/macos/AdaptiveCards/AdaptiveCards/Renderers/InputNumberRenderer.swift [hunk 1]
```diff
@@ -34,24 +33,31 @@ open class InputNumberRenderer: NSObject, BaseCardElementRendererProtocol {
...
+        let view = ACRTextField(numericFieldWith: config)
         view.translatesAutoresizingMaskIntoConstraints = false
         view.delegate = self
         view.isEditable = true
+        view.stringValue = ""
+        if #available(OSX 10.13, *) {
+            view.layer?.maskedCorners = [.layerMinXMaxYCorner, .layerMinXMinYCorner]
+        } else {
+            // Do nothing as UX Rebrand only part of 41.11 which has minimum requirement of OSX 10.13
+        }
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Renderers/InputNumberRenderer.swift:60]: We can avoid comments related to "Webex" in this library.

File: source/macos/AdaptiveCards/AdaptiveCards/Renderers/InputNumberRenderer.swift [hunk 2]
```diff
@@ -66,18 +72,30 @@ open class ACRNumericTextField: NSView, NSTextFieldDelegate {
...
-        stepper.doubleValue = Double(textView.accessibilityValue() ?? "") ?? stepper.doubleValue
-        guard let stringValue = textView.accessibilityValue() else { return }
-        if Double(stringValue) != nil {
-            textView.stringValue = stringValue
-        } else if Int(stringValue) == nil && !stringValue.isEmpty {
-            textView.stringValue = previousValue
-        } else if stringValue.isEmpty {
-            textView.stringValue = ""
+        if let textfield = obj.object as? ACRTextField {
+            var stringValue = textfield.stringValue
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Renderers/InputNumberRenderer.swift:76]: Prefer `guard let` here

File: source/macos/AdaptiveCards/AdaptiveCards/Renderers/InputNumberRenderer.swift [hunk 3]
```diff
@@ -66,18 +72,30 @@ open class ACRNumericTextField: NSView, NSTextFieldDelegate {
...
+            
+            stepper.doubleValue = Double(textField.accessibilityValue() ?? "") ?? stepper.doubleValue
+            let charSet = NSCharacterSet(charactersIn: "1234567890.").inverted
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Renderers/InputNumberRenderer.swift:79]:
```suggestion
            stepper.doubleValue = Double(textField.stringValue ?? "") ?? stepper.doubleValue
            let charSet = NSCharacterSet(charactersIn: "1234567890.").inverted
```
`accessibilityValue()` might different purposes to solve

File: source/macos/AdaptiveCards/AdaptiveCards/Renderers/InputNumberRenderer.swift [hunk 4]
```diff
@@ -177,7 +180,11 @@ extension ACRNumericTextField: InputHandlingViewProtocol {
     }
     
     var value: String {
-        return String(Int(inputValue))
+        if Float(inputValue).truncatingRemainder(dividingBy: 1) == 0 {
+            return String(Int(inputValue))
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Renderers/InputNumberRenderer.swift:184]: Isn't this same as Int(inputValue) ??

File: source/macos/AdaptiveCards/AdaptiveCards/Views/ACRTextField.swift [hunk 5]
```diff
@@ -10,11 +10,13 @@ class ACRTextField: NSTextField {
     private let config: InputFieldConfig
     private let isDarkMode: Bool?
     private let isDateTimeMode: Bool
+    private var isNumericField: Bool
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Views/ACRTextField.swift:14]:
```suggestion
    private let isNumericField: Bool
    
```

Comment (blue-stone99) [reply source/macos/AdaptiveCards/AdaptiveCards/Views/ACRTextField.swift:14]: Maybe we should think about introducing `enum TextFieldMode` with `dateTime`, `numeric`, `normal` and maintain a single mode property here.

File: source/macos/AdaptiveCards/AdaptiveCards/Renderers/InputNumberRenderer.swift (hunk 4 above)
Comment (mukuagar) [reply source/macos/AdaptiveCards/AdaptiveCards/Renderers/InputNumberRenderer.swift:184]: Int(12.3) returns 12, so I was doing this to make sure the number is a perfect integer

File: source/macos/AdaptiveCards/AdaptiveCards/Views/ACRTextField.swift [hunk 6]
```diff
@@ -5,15 +5,23 @@ protocol ACRTextFieldDelegate: AnyObject {
     func acrTextFieldDidSelectClear(_ textField: ACRTextField)
 }
 
+private enum textFieldMode {
+    case inputText
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Views/ACRTextField.swift:9]: 👍

Comment (blue-stone99) [reply source/macos/AdaptiveCards/AdaptiveCards/Views/ACRTextField.swift:9]:
```suggestion
private enum Mode {
    case inputText
```

This enum can be inside `ACRTextField` class.

File: source/macos/AdaptiveCards/AdaptiveCards/Views/ACRTextField.swift [hunk 7]
```diff
@@ -5,15 +5,23 @@ protocol ACRTextFieldDelegate: AnyObject {
...
+    case inputNumber
+}
+
 class ACRTextField: NSTextField {
     weak var textFieldDelegate: ACRTextFieldDelegate?
     private let config: InputFieldConfig
     private let isDarkMode: Bool?
-    private let isDateTimeMode: Bool
+//    private let isDateTimeMode: Bool
+//    private var isNumericField: Bool
```
Comment (blue-stone99) [inline source/macos/AdaptiveCards/AdaptiveCards/Views/ACRTextField.swift:19]:
```suggestion
```

Can be removed.

**Pull request merged**